### DISCUSSION TITLE
What JRE should I use for the new Air SDK 50.2 on Flash Builder 4.7

### DISCUSSION THREAD

**Discussion by danielkotzer** [airsdk/Adobe-Runtime-Support#2484] [General]
Hi,

When I was working with Air SDK version 33.1... I had a problem updating the JRE folder in Flash Builder 4.7 to a higher version then of jre1.8.0_112, which can be found in the installation folder of jdk-8u112-windows-x64.exe or else I would get problems when creating a build. But now that a new Air SDK is out, do I have to update the JRE file on the Flash Builder to a higher version of JER?
And if I do, then to what version? the latest versions of JDK don't contain a JRE folder anymore, from version 11 and higher, so what java version do I have to take the JRE folder from? And is there a problem uninstalling this version of Java from my Windows 10, after I took out the jre folder and copied it to the FB folder, and installing the latest version of java jdk, or will this create conflicts?

Thanks

**Comment by ajwfrost** [comment]
You should be able to leave Flash Builder with the 1.8.0_112 version of the Java runtime, the AIR tools are all compatible with this. The only issue you may have if using Android is that the Android build tools require Java 11 now I think; if you set the "JAVA_HOME" value in your AIR configuration file (or via AIR SDK Manager) then the AIR Developer Tool (running in Java 8) should launch a separate process for Gradle (sing Java 11).

If there's no "jre" folder in the JDK then I would assume you just copy the jdk itself, I hadn't noticed that change. Mostly it would just look in the "bin" folder for a file "java.exe"..

If you copy from an installation, and paste the files somewhere else, then those won't affect the installation -> you'll still be able to update/uninstall the original package and it will affect (only) the original location of the files. So copying and pasting from an installed JDK or JRE into the Flash Builder folder is fine in terms of the Java installers! If you ever ran the FB uninstaller, you'd find that it won't clean up those copied JRE files..

thanks
